### DISCUSSION TITLE
NVT, CERT, SCAP, GVMD Cache

### DISCUSSION THREAD

**Discussion by mrod23** [Secure-Compliance-Solutions-LLC/GVM-Docker#189] [General]
Can you share the script you run to cache the nvt, cert, scap, gvmd data?
The data that you curl here: https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/blob/6b7d55ed6dfbed1f7ec2360feaf524bc58f0aceb/scripts/sync-all.sh#L5

Also, would you be opposed to setting an environment variable for the below link, so we can use our own cache?
https://vulndata.securecompliance.solutions/file/VulnData/data.tar.xz

I can submit a pull request for that.

**Comment by austinsonger** [comment]
@pixelsquared

**Reply by mrod23** [reply]
I took another look at the sync-all script over the weekend. I would like to update if you don't have any issues. 

First I want to explain my use case for the Openvas containers. 
I'm currently scanning thousands of IPs for my company. In order to do that I need to have multiple OpenVas containers spun up. 
When I run multiple OpenVas containers from the same Kubernetes cluster, they fail at the sync-all step in the start.sh file, because they have the same external IP and are blocked by OpenVas for pulling down the feeds more than once per day from the same IP. 

I'd like to update the sync script so that we can use a custom environment variable that points to a cache that has all the feed data. If the cache fails, the script can revert to running the greenbone-nvt-feed sync commands. 

Let me know what you think. Thank you.

**Comment by pixelsquared** [comment]
We are using github actions in a private repo for the data sync I am planing on making that repo public shortly.

**Reply by mrod23** [reply]
Any luck on this?
Also I think the current sync script doesn't pull in all data from your cache here: https://vulndata.securecompliance.solutions/file/VulnData/data.tar.xz

For example, when I run the sync-all script and only pull in the cached data I'm missing report_formats.

But when I run `greenbone-nvt-sync` I get report_formats

**Reply by mrod23** [reply]
If there's something I can do to help out with the sync scripts, please let me know. 
I used to have my own cache using the old rsync commands. The OpenVas upgrades broke my scripts. 

I used to be able to run this python function `report = gmp.get_report(report_id=task_report_id, details=True, report_format_id="c1645568-627a-11e3-a660-406186ea4fc5" )`

This doesn't work anymore because report formats have to be imported via greenbone-nvt-sync

If I have the container spin up more than once per day, greenbone will block my sync

**Reply by mrod23** [reply]
Or if you know a way to import report_formats, that could work also. 
Thanks in advance

**Reply by pixelsquared** [reply]
Hi,

I do know why the report formats are not importing for you if you are not using the Greenbone sync script. 

The data file from https://vulndata.securecompliance.solutions/file/VulnData/data.tar.xz is missing a few files that are generated by the Greenbone sync script. I will work on adding the missing data to the archive and releasing the repo.

**Reply by mrod23** [reply]
Thanks @pixelsquared appreciate that

**Reply by mrod23** [reply]
@pixelsquared  do you know when you'll have the update in by?
Also, let me know if I can help at all with the PR.

**Reply by austinsonger** [reply]
@mrod23  
If you want to contribute by PRs, then please read this discussion I created. 
- https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/discussions/193

**Reply by mrod23** [reply]
Added my info

**Reply by austinsonger** [reply]
@mrod23 take a look at issues in this milestone https://github.com/Secure-Compliance-Solutions-LLC/GVM-Docker/milestone/16
